core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1365,19 +1369,27 @@ class DAGScheduler(
    */
   private[scheduler] def handleExecutorLost(
       execId: String,
-      filesLost: Boolean,
+      fileLost: Boolean,
+      hostLost: Boolean = false,
+      maybeHost: Option[String] = None,
```

Comment:
I find this method pretty confusing now, but it was also confusing before, and I'm not sure how to clean it up yet.  one minor thing: instead of having a `hostLost` and `maybeHost`, could there be a `hostToDeregisterAllShuffleOutput: Option[String]`, and you replace `if (hostLost) {...}` with `hostToDeregisterAllShuffleOutput.foreach{...}` etc.